.tekton/tasks/task-deploy-to-code-engine.yml

```diff
@@ -0,0 +1,209 @@
+    - name: code-engine-project
+      description: name of the Code Engine project for deploying application
+    - name: service-bindings
+      description: JSON object of a map of names to services instances to be bound to the application
+      default: "{}"
+    - name: image-repository
+      description: the repository for the built image
+    - name: image-tags
+      description: the tags for the built image
+    - name: shuttle-properties-file
+      description: name of the properties file that contain properties to include in the environment
+      default: "build.properties"
+    - name: pipeline-debug
+      description: Pipeline debug mode. Value can be 0 or 1. Default to 0
+      default: "0"
+  results:
+    - name: app-url
+      description: The running application's URL (obtained from APP_URL variable set by the executed script)
+  workspaces:
+    - name: artifacts
+      description: A workspace
+      mountPath: /artifacts
+  steps:
+    - name: execute
+      image: ibmcom/pipeline-base-image:2.12
+      workingDir: /artifacts
+      env:
+        - name: BUILD_NUMBER
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.annotations['devops.cloud.ibm.com/build-number']
+        - name: PIPELINE_ID
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.annotations['devops.cloud.ibm.com/pipeline-id']
+        - name: PIPELINE_BLUEMIX_API_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.continuous-delivery-context-secret)
+              key: $(params.ibmcloud-apikey-secret-key)
+        - name: SHUTTLE_PROPERTIES_FILE
+          value: $(params.shuttle-properties-file)
+      script: |
+        #!/bin/bash
+        set -e -o pipefail
+
+        if [ $(params.pipeline-debug) == 1 ]; then
+          env
+          pwd
+          ls -l
+          echo "=== cat /artifacts/_toolchain.json ==="
+          cat /artifacts/_toolchain.json
+          echo ""
+          echo "======================================"
+          trap env EXIT
+          set -x
+        fi
+
+        # SETUP BEGIN
+        ibmcloud config --check-version false
+        if [ "$(params.region)" ]; then
+          # if cluster region is in the 'ibm:yp:<region>' just keep the region part
+          IBM_CLOUD_REGION=$(echo "$(params.region)" | awk -F ':' '{print $NF;}')
+        else
+          IBM_CLOUD_REGION=$(jq -r '.region_id' /artifacts/_toolchain.json | awk -F: '{print $3}')
+        fi
+
+        ibmcloud login -a $(params.ibmcloud-api) -r $IBM_CLOUD_REGION --apikey $PIPELINE_BLUEMIX_API_KEY
+
+        ibmcloud target -g $(params.resource-group)
+
+        # View shuttle properties
+        cat $(params.shuttle-properties-file)
+
+        source $(params.shuttle-properties-file)
+        export $(cut -d= -f1 $(params.shuttle-properties-file))
+
+        ibmcloud plugin install code-engine
```

Comment:
Not a requirement, but might want to consider a custom image or have DevOps add the plugin to the base image so that we are not installing this every time. Should open an issue for that.